pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -573,6 +591,18 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 					StartupProbe:    controller.GenerateStartupProbeExecDaemon(opconfig.OsdType, osdID),
 					LivenessProbe:   controller.GenerateLivenessProbeExecDaemon(opconfig.OsdType, osdID),
 					WorkingDir:      opconfig.VarLogCephDir,
+					Ports: []v1.ContainerPort{
+						{
+							Name:          "osd-port",
+							ContainerPort: 6800,
+							Protocol:      v1.ProtocolTCP,
+						},
+						{
+							Name:          "osd-port-2",
```

Comment:
Need to check again with Radoslaw why 6801 port is needed.

Reply:
https://www.sebastien-han.fr/blog/2012/07/29/tip-ceph-public-slash-private-network-configuration/ says its for internal OSD communication for replication

Reply:
6801 is for the old ProtocolV1 which is still being used by older kernels (krbd)

Reply:
If only msgr2 is enabled, let's skip adding the v1 port. You can check this with the `RequireMsgr2()` helper.